contracts/core/TribeAuthority.sol

```diff
@@ -0,0 +1,37 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.8.0;
+
+import "../external/solmate/src/auth/authorities/MultiRolesAuthority.sol";
+
+/// @notice Admin based Authority for a MultiRolesAuthority
+abstract contract TribeAuthority is MultiRolesAuthority {
+
+    mapping(uint8 => bytes32) public getRoleAdmins;
+
+    // TODO add a way to assign admins and a constructor
+    
+    /*///////////////////////////////////////////////////////////////
+                      USER ROLE ASSIGNMENT LOGIC
+    //////////////////////////////////////////////////////////////*/
+
+    function isRoleAdmin(address admin, uint8 role) public view returns(bool) {
+        return (getRoleAdmins[role] & getUserRoles[admin]) != bytes32(0);
```

Comment:
Why bitwise AND here?

Reply:
solmate uses bit flags and shifting for storing info about user roles, so it's checking to see if a user has an admin role by checking to see if they both have that same bit flipped